gateway/ln-gateway/src/rpc/rpc_server.rs

```diff
@@ -144,3 +180,27 @@ async fn restore(
     gateway.handle_restore_msg(payload).await?;
     Ok(())
 }
+
+#[instrument(skip_all, err)]
+async fn set_configuration_public(
+    Extension(sender): Extension<mpsc::Sender<GatewayConfiguration>>,
+    Json(payload): Json<SetConfigurationPayload>,
+) -> Result<impl IntoResponse, GatewayError> {
+    let gateway_configuration = GatewayConfiguration {
+        password: payload.password.clone(),
+    };
+
+    sender.send(gateway_configuration).await.map_err(|e| {
+        GatewayError::UnexpectedState(format!("Gateway is not in Configuring state: {e:?}"))
+    })?;
```

Comment:
Have we considered if there's any configuration we would want to change after initial setup? If you want to change something while your gateway is already running, how would we do that?

Reply:
This should be supported. There's a different `set_configuration` endpoint on the webserver that runs after the config is set as well, which writes the config to the database.

The tricky part is applying the change after it is set in the database. Other operations (like fees) should be able to just query the db to pickup the new change. For the gateway's password though, we need to restart the webserver, which seemed like too much to implement for this PR.